src/validate.ts

```diff
@@ -0,0 +1,79 @@
+'use strict';
+
+chai.use(require('chai-json-schema-ajv'));
+
+function generateRandom(min, max){
+    //helper function for obtaining a random requestId
+    return Math.floor(Math.random() * (max-min) + min).toString();
+}
+
+
+function generateSyncReq(){
+    const requestId = generateRandom(100,999);
+    return {
+        requestId,
+        "inputs": [{
+          "intent": action.devices.SYNC
+            }]
+    }
+}
+
+const sync_request = generateSyncReq();
+const requestRandId = sync_request["requestId"];
+
+function generateQueryReq(deviceID){
+    
+    const intent = "action.devices.QUERY";
+    var queryReq = {};
+    queryReq.requestId = requestRandId;
+    queryReq.inputs = [];
+    queryReq.inputs.unshift({});
+    queryReq.inputs[0]["intent"] = intent;
+    queryReq.payload = {};
+    queryReq.payload.devices = [];
+    queryReq.payload.devices.unshift({})
+    queryReq.payload.devices[0]["id"] = deviceID;
+
+    
+    //  for (var i = 0; i <deviceID.length; i++){
+    //         queryReq.payload.devices[i]["id"] = deviceID[i];
+    //     }
+    //add custom data
+    
+    return queryReq;
+    
+    
+}
+
+const deviceInfo = (query_request["inputs"][0]["payload"]);
+
+function generateExecuteReq(traits, params){
```

Comment:
The way this is currently used in the function, `traits` is actually the name of a single command rather than a specific trait.